src/bindings/python/tests/test_runtime/test_infer_request.py

```diff
@@ -475,12 +483,17 @@ def callback(request, job_id):
         jobs_done[job_id]["finished"] = True
         jobs_done[job_id]["latency"] = request.latency
 
-    img = generate_image()
+    img = None
+
+    if not shared_flag:
+        img = generate_image()
     infer_queue.set_callback(callback)
     assert infer_queue.is_ready()
 
     for i in range(jobs):
-        infer_queue.start_async({"data": img}, i)
+        if shared_flag:
+            img = generate_image()
```

Comment:
Could you please explain why a new `img` needs to be created for every async if `shared_memory=True`? Would it matter if the random image seed is constant? Or is it about asserting that the original `img` object is still alive after overwriting it in the next loop iterations? Not a change request, just want to understand what's going on :)

Reply:
Discussed in details offline. Short note: due to potential memory conflicts/data-races, there must should be separated images.